prompt_toolkit/input/win32.py

```diff
@@ -260,7 +263,7 @@ def read(self) -> Iterable[KeyPress]:
                 # Pasting: if the current key consists of text or \n, turn it
                 # into a BracketedPaste.
                 data = []
-                while k and (isinstance(k.key, str) or k.key == Keys.ControlJ):
+                while k and (not isinstance(k.key, Keys) or k.key == Keys.ControlJ):
```

Comment:
Is this change needed?

Reply:
This is unrelated to the main change but a potential bug fix. Because the `Keys` enum inherits `str`, technically the original check would always pass, so I think this change more correctly matches the intent of the code. Actually it would probably be better to make the enum not inherit `str` in the first place (which makes the enum a drop-in replacement for str in most cases); I don't think this is very useful but it may be relied on by some code so not sure if it would be safe to change without analyzing more of the code base.